jobs/pulse3d/test-pulse3d-worker/src/main.py

```diff
@@ -0,0 +1,360 @@
+                    param: val
+                    for param in (
+                        "relative_prominence_factor",
+                        "noise_prominence_factor",
+                        "height_factor",
+                        "width_factors",
+                        "max_frequency",
+                        "valley_search_duration",
+                        "upslope_duration",
+                        "upslope_noise_allowance_duration",
+                    )
+                    if (val := analysis_params.get(param)) is not None
+                }
+                logger.info("Running peak detector")
+                data_with_features = peak_finder.run(windowed_pre_analyzed_data, alg_args=peak_detector_args)
+            except Exception:
+                logger.exception("PeakDetector failed")
+                raise
+
+            try:
+                # TODO sync these values
+                # twitch_widths v. widths --- baseline_widths_to_use v. baseline_widths
+                metrics_args = {
+                    arg_name: val
+                    for arg_name, orig_name in (
+                        ("widths", "twitch_widths"),
+                        ("baseline_widths", "baseline_widths_to_use"),
+                        ("well_groups", "well_groups"),
+                    )
+                    if (val := analysis_params.get(orig_name)) is not None
+                }
+                logger.info("Running metrics")
+                metrics_output = metrics.run(data_with_features, **metrics_args)
+            except Exception:
+                logger.exception("Metrics failed")
+                raise
+
+            try:
+                renderer_args = {
+                    arg_name: val
+                    for arg_name in (
+                        "include_stim_protocols",
+                        "stim_waveform_format",
+                        "data_type",
+                        "normalize_y_axis",
+                        "max_y",
+                    )
+                    if (val := analysis_params.get(arg_name)) is not None
+                }
+
+                logger.info("Running renderer")
+                output_filename = renderer.run(
+                    metrics_output, OutputFormats.XLSX, output_format_args=renderer_args
+                )
+            except Exception:
+                logger.exception("Renderer failed")
+                raise
+
+            try:
+                outfile_prefix = prefix.replace("uploads/", "analyzed/test-pulse3d/")
+                outfile_key = f"{outfile_prefix}/{job_id}/{output_filename}"
+                upload_file_to_s3(bucket=PULSE3D_UPLOADS_BUCKET, key=outfile_key, file=output_filename)
+                logger.info(f"Uploaded {output_filename} to {PULSE3D_UPLOADS_BUCKET}/{outfile_key}")
+            except Exception:
+                logger.exception("Upload failed")
+                raise
+
+            if not interactive_analysis:
```

Comment:
We might want to upload all this intermediate data right after it's done so that if a downstream module errors we don't skip uploading to S3

Reply:
yeah I like that